Minor linguistic changes (#600)
ezsystems/developer-documentation: docs/guide/persistence_cache.md

Persistence cache can best be described as an implementation of `SPI\Persistence` that decorates the main backend implementation, aka Storage Engine *(currently: "Legacy Storage Engine")*.

As shown in the illustration, this is done in the exact same way as the SignalSlot feature is a custom implementation of `API\Repository` decorating the main Repository. 
In the case of Persistence Cache, instead of sending events on calls passed on to the decorated implementation, most of the load calls are cached, and calls that perform changes purge the affected caches.
Cache handlers *(Memcached, Redis, Filesystem, etc.)* can be configured using Symfony configuration. 
For details on how to reuse this Cache service in your own custom code, see below.

## Transparent cache

> With the persistence cache, just like with the HTTP cache, eZ Platform tries to follow principles of "Transparent caching".

Reviewer: ```suggestion
With the persistence cache, just like with the HTTP cache, eZ Platform tries to follow principles of transparent caching.
```

This can shortly be described as a cache which is invisible to the end user (admin/editors) of eZ Platform where content

> is always returned "fresh." In other words, there should be no need to manually clear the cache like it was frequently

Reviewer: ```suggestion
is always returned *fresh*. In other words, there should be no need to manually clear the cache like it was frequently
```

Notes:

- `UrlWildCardHandler` is not currently cached.
- Currently in case of transactions this is handled by clearing all cache on rollback, so avoid using rollbacks
  as part of your normal application logic flow. For instance if you connect to third party service and it frequently
  fails, make sure to re-try several times, and if that does not help, consider making the logic async by design.
- [Cache tagging](https://symfony.com/doc/current/components/cache/cache_invalidation.html#using-cache-tags) is used in
  order to allow clearing cache by alternative indexes. 

>   For instance tree operations or changes to content types are

Reviewer: ```suggestion
  For instance tree operations or changes to Content Types are
```

  examples of operations that also need to invalidate content cache by tags.
- Search is not defined as persistence and the queries themselves are not planned to be cached as they are too complex by design (full text, facets, etc.).
  Use [Solr](solr.md) which caches this for you to improve scale, and to offload your database.

> *For further details on which calls are cached or not, and where/how to contribute additional caches, go through the [source](https://github.com/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/Core/Persistence/Cache).*

Reviewer: ```suggestion
*For further details on which calls are cached or not, and where/how to contribute additional caches, see the [source](https://github.com/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/Core/Persistence/Cache).*
```

!!! caution "Use unique vendor prefix for Cache key"

    When reusing the cache service within your own code, it is very important to not conflict with the cache keys used by others.
    That is why the example of usage below starts with a unique `myApp` key.

>     For the namespace of your own cache, you **must** do the same.

Reviewer: ```suggestion
    For the namespace of your own cache, you must do the same.
```
